src/api/platform/models/api_drive_access.rs

```diff
@@ -0,0 +1,49 @@
+use super::{ApiDriveId, ApiKeyId};
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug, Deserialize, Serialize)]
+#[cfg_attr(feature = "strict", serde(deny_unknown_fields))]
+pub struct ApiDriveAccess {
+    user_key_id: ApiKeyId,
+    #[serde(rename = "bucket_id")]
+    drive_id: ApiDriveId,
+    fingerprint: String,
+    state: BucketAccessState,
+}
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum BucketAccessState {
+    Pending,
+    Approved,
+    Revoked,
```

Comment:
Do we ever return a revoked key? I thought we were just deleting associations rather than including them in any kind of a list